### PR TITLE
Iana files revisited

### PR DESCRIPTION
**Pull request by wibisono**
Changes:
* Adjust column to add status (reserved, allocated, available) according to [the latest](http://ftp.apnic.net/stats/iana/delegated-iana-latest) IANA file.
* Include [ASN](https://www.iana.org/assignments/iana-as-numbers-special-registry/iana-as-numbers-special-registry.xhtml) and [IPv6 Special Registry](https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml)
* Self documenting variable names for  [hard-coded dates](https://github.com/RIPE-NCC/nro-delegated-stats/blob/iana-revisited/src/main/scala/net/ripe/rpki/nro/Const.scala#L45)
* Added a separate commands `iana-file` to generate the IANA file without using it on generation of delegated stats.
* Difference with the latest IANA file are mostly on dates, ipv6 formatting (:0000 vs ::), and status of `allocated|ietf` vs `reserved|ietf`.
* Testing this implementation can be done by generating the IANA file:
```
 sbt assembly && java -jar target/scala-2.13/nro-delegated-stats.jar iana-file
 ```
* A helpful shell command to skip the date columns when comparing difference with original iana file:
```
diff <(cat result/20220930/delegated-iana | cut -d'|' -f1,2,3,4,5,7,8) <(cat data/20220930/iana | cut -d'|' -f 1,2,3,4,5,7,8)
```

### CODE REVIEW

File: src/main/scala/net/ripe/rpki/nro/service/Ports.scala [hunk 1]
```diff
@@ -199,15 +193,20 @@ object Ports extends Logging {
 
 
 
-  def writeRecords(records: Records, outputFile: String = s"$resultFileName", header: Boolean = true): Unit = {
-      writeResult(records.asn, records.ipv4, records.ipv6, outputFile, header)
+  def writeRecords(records: Records, outputFile: String = s"$resultFileName", header: Boolean = true, disclaimer: Boolean = false): Unit = {
```
Comment (bjpbakker) [inline src/main/scala/net/ripe/rpki/nro/service/Ports.scala:196]: Prefer composition over passing around booleans for special cases

**Review by bjpbakker — CHANGES_REQUESTED** [review]
Some things I think can be improved. Need to test the full process still.

File: src/main/resources/disclaimer.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
+######################################################################
+#
+#       CONDITIONS OF USE
```
[the remark this answers is not in the document]
Comment (wibisono) [reply src/main/resources/disclaimer.txt:3]: This has been updated to RIPE's (had asked registry team which to use yesterday)
https://github.com/RIPE-NCC/nro-delegated-stats/pull/20/commits/cd54934cef16e0bca3b87915635a30383dd54b05


I'll run the text through legal.

File: src/main/scala/net/ripe/rpki/nro/main/Ranges.scala [hunk 3]
```diff
@@ -32,9 +32,13 @@ trait Ranges {
...
    *         Align
    */
-  def alignRecordWithMapRangeKeys(mapToAlign: RangeMap[BigInteger, Record]): List[Record] = mapToAlign.asMapOfRanges().asScala.toList
+  def alignRecordWithMapRangeKeys(mapToAlign: RangeMap[BigInteger, Record], alignIpv4: Boolean = false): List[Record] = mapToAlign.asMapOfRanges().asScala.toList
     .flatMap {
       case (key, _) if empty(key) => List()
+      case (range, record: Ipv4Record) if alignIpv4 =>
+        record.splitPrefixes(range).map { ipv6 =>
+          record.updateRange(RecordRange.from(ipv6).key)
+        }
```
[the remark this answers is not in the document]
Comment (wibisono) [reply src/main/scala/net/ripe/rpki/nro/main/Ranges.scala:41]: Indeed, resolved

File: src/main/scala/net/ripe/rpki/nro/service/Ports.scala [hunk 4]
```diff
@@ -254,7 +253,7 @@ object Ports extends Logging {
         .sliding(2, 2)
         .map {
           case Seq(a, b) => Conflict(Record(a), Record(b))
-          case x => throw new IllegalArgumentException("Cannot parse incomplete conflicts file. Remainder: $x")
+          case _ => throw new IllegalArgumentException("Cannot parse incomplete conflicts file. Remainder: $x")
```
[the remark this answers is not in the document]
Comment (wibisono) [reply src/main/scala/net/ripe/rpki/nro/service/Ports.scala]: Done.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -81,3 +81,46 @@ jobs:
 
       - name: Image digest
         run: echo ${{ steps.docker_build_nro_delegated_stats.outputs.digest }}
+
+  docker-branch:
```
[the remark this answers is not in the document]
Comment (wibisono) [reply .github/workflows/ci.yml]: Done

File: src/main/scala/net/ripe/rpki/nro/service/Ports.scala (hunk 1 above)
Comment (wibisono) [reply src/main/scala/net/ripe/rpki/nro/service/Ports.scala]: Refactored using composition.

**Review by bjpbakker — APPROVED** [review]

**Pull request merged**